python/pynessie/cli_common_context.py

```diff
@@ -77,3 +77,24 @@ def parse_args(self: "DefaultHelp", ctx: click.Context, args: List) -> List:
 
 
 pass_client = click.make_pass_decorator(ContextObject)
+
+
+def validate_options(f: Callable) -> Callable:
+    """This Decorators is responsible to validate CLI options and override them with default if needed."""
+
+    @functools.wraps(f)
+    def wrapper(*args: Any, **kwargs: Any) -> None:
+        """Wrapper Decorator."""
+        # Nessie context is always as the first argument
+        if len(args) < 1 or not isinstance(args[0], ContextObject):
+            raise RuntimeError("Use this decorator only with Nessie context.")
+
+        ctx: ContextObject = args[0]
+
+        # Check for ref and override with default if is not set
+        if "ref" in kwargs:
+            kwargs["ref"] = kwargs["ref"] if kwargs["ref"] else ctx.nessie.get_default_branch()
```

Comment:
Yes it can be `None` in case the user doesn't supply `--ref` flag. I think IMO it won't make sense to raise exception since we want to fallback to the default branch in case the user didn't specify the branch.

Reply:
but if `--ref` is not set on cmd. line, would this `if "ref" in kwargs` not be `False`?

Reply:
`if "ref" in kwargs` would check list of arguments that set by `@click.option` or `@click.argument`. So regardless if `--ref` is set from the command line or not, as long is added through `@click.option`, it will be present but with value of `kwargs["ref"]=None`